submissions/vakulinina/friends-app/main.js

```diff
@@ -0,0 +1,174 @@
+  }
+};
+
+const sortByAge = users => {
+  if (filter.ageSort === ASCENDING_ORDER) {
+    return users.sort((a, b) => compareByAge(a, b))
+  } else {
+    return users.sort((a, b) => compareByAge(b, a))
+  }
+};
+
+const filterByAge = users => {
+  const [ageFrom, ageTo] = [filter.ageFrom, filter.ageTo].sort((a, b) => a - b);
+  return users.filter(({ dob: { age } }) => (age >= ageFrom && age <= ageTo))
+};
+
+const filterByGender = users => {
+  return users.filter(user => user.gender === filter.gender)
+};
+
+const doSearch = users => {
+  return users.filter(({ name: { first, last }, location: { city } }) => {
+    const fullName = `${first.toLowerCase()} ${last.toLowerCase()}`;
+    return fullName.includes(filter.searchValue) || city.toLowerCase().includes(filter.searchValue)
+  })
+};
+
+const filterUsers = () => {
+  let filteredUsers = [...allUsers];
+  if (filter.ageSort) {
+    filteredUsers = sortByAge(filteredUsers);
+  }
+  if (filter.nameSort) {
+    filteredUsers = sortByName(filteredUsers);
+  }
+  if (filter.ageFrom || filter.ageTo) {
+    filteredUsers = filterByAge(filteredUsers);
+  }
+  if (filter.gender !== 'any') {
+    filteredUsers = filterByGender(filteredUsers);
+  }
+  if (filter.searchValue) {
+    filteredUsers = doSearch(filteredUsers);
+  }
+  renderUsers(filteredUsers)
+};
+
+const setNameSortOrder = () => {
+  if (filter.nameSort === ASCENDING_ORDER) {
+    filter.nameSort = DESCENDING_ORDER;
+    sortByNameButton.classList.replace(ASCENDING_ORDER, DESCENDING_ORDER)
+  } else {
+    filter.nameSort = ASCENDING_ORDER;
+    sortByNameButton.classList.remove(DESCENDING_ORDER)
+    sortByNameButton.classList.add(ASCENDING_ORDER)
+  }
+  sortByAgeButton.classList.remove(ASCENDING_ORDER, DESCENDING_ORDER)
+  filter.ageSort = null;
+};
+
+const setAgeSortOrder = () => {
+  if (filter.ageSort === ASCENDING_ORDER) {
+    filter.ageSort = DESCENDING_ORDER;
+    sortByAgeButton.classList.replace(ASCENDING_ORDER, DESCENDING_ORDER)
+  } else {
+    filter.ageSort = ASCENDING_ORDER;
+    sortByAgeButton.classList.remove(DESCENDING_ORDER)
+    sortByAgeButton.classList.add(ASCENDING_ORDER)
+  }
+  sortByNameButton.classList.remove(ASCENDING_ORDER, DESCENDING_ORDER)
+  filter.nameSort = null;
+};
+
+const handleInput = (target) => {
+  const filters = {
+    'name-sort': setNameSortOrder,
+    'age-sort': setAgeSortOrder,
+    'age-from': () => filter.ageFrom = target.value || MIN_AGE,
+    'age-to': () => filter.ageTo = target.value || MAX_AGE,
+    'search-field': () => filter.searchValue = target.value.toLowerCase()
+  };
+
+  filters[target.id] ? filters[target.id]() : filter.gender = target.value;
+  filterUsers();
+};
+
+const initialize = async () => {
+  const filtersBar = document.querySelector('.filters');
+  await getUsers()
+  renderUsers(allUsers)
+  filtersBar.addEventListener('input', ({ target }) => handleInput(target));
+  document.querySelector('.sort-buttons').addEventListener('click', ({ target }) => {
```

Comment:
Buttons can be inputs and then they will fire `change` event and you will have a possibility to remove check at the next line

Reply:
I'm stuck here. Do you mean `<input type="button">`? How can I make it fire `change` event? It doesn't if I click it.

Reply:
As an example https://markheath.net/post/customize-radio-button-css